api-controllers/ServiceFabrikAdminController.js

```diff
@@ -52,19 +68,41 @@ class ServiceFabrikAdminController extends FabrikBaseController {
           });
         });
     }
-    logger.info(`Forbidden Manifest flag set to ${allowForbiddenManifestChanges}`);
-    if (allowForbiddenManifestChanges === false) {
-      const instanceId = this.parseServiceInstanceIdFromDeployment(deploymentName);
-      return this
-        .getServicePlanIdForInstanceId(instanceId)
-        .then(planId => this.getOutdatedDiff(deploymentName, catalog.getPlan(planId)))
-        .then(diff => utils.hasChangesInForbiddenSections(diff))
-        .tap(() => logger.info(`Doing update for ${deploymentName} as there is no forbidden changes in manifest`))
-        .then(() => updateDeployment());
-    } else {
-      logger.info(`Doing update for ${deploymentName} even if forbidden changes exist in manifest`);
-      return updateDeployment();
-    }
+
+    return Promise.try(() => {
+      logger.info(`Forbidden Manifest flag set to ${allowForbiddenManifestChanges}`);
+      return eventmesh.apiServerClient.getResource({
+          resourceGroup: CONST.APISERVER.RESOURCE_GROUPS.DEPLOYMENT,
+          resourceType: CONST.APISERVER.RESOURCE_TYPES.DIRECTOR,
+          resourceId: instanceId
+        })
+        .catch(errors.NotFound, () => undefined)
+        .then(resource => _.get(resource, 'spec.options'))
+        .then(resource => {
+          resourceDetails = resource;
+          if (resourceDetails === undefined) {
+            throw new errors.NotFound(`Resource details of service instance ${instanceId} not found in api server.`);
```

Comment:
Can we have this method segregated to find Plan_ID and organization guid and space guid.

Reply:
Well, can be. How to return? as composite obj ? Then use like obj.plan, obj.organization_guid, obj.space_guid?
Might be required to fetch some other parameters. So as *resourceDetails* having all, ad-hoc facing would be good? or should I change ?

Reply:
recource details should be fine.